src/include/buffer.h

```diff
@@ -334,6 +334,9 @@ namespace buffer CEPH_BUFFER_API {
 
     unsigned append(char c);
     unsigned append(const char *p, unsigned l);
+    inline unsigned append(std::string_view s) {
```

Comment:
this breaks the support of C++ compiler/libstdc++/libc++ which does not support C++17, also this header is not self-contained. probably we can have something like
```c++
#if _cplusplus >= 201703L
#include <string_view>
#endif

// ...
#if _cplusplus >= 201703L
inline unsigned append(std::string_view s) {
  return append(s.data(), s.length());
}
#endif
```

see http://pulpito.ceph.com/kchai-2018-02-07_01:22:25-rados-wip-kefu-testing-2018-02-06-1514-distro-basic-mira/2161239/